src/allmydata/immutable/downloader/finder.py

```diff
@@ -63,6 +63,7 @@ def start_finding_servers(self):
         if not self._started:
             si = self.verifycap.storage_index
             servers = self._storage_broker.get_servers_for_psi(si)
+            servers.sort(key=lambda s: s.get_serverid())
```

Comment:
Hm, removing it causes `test_hung_server.HungServerDownloadTest.test_5_overdue_immediate` to fail consistently. I'll look into that.